general/tassign.hpp

```diff
@@ -43,70 +43,110 @@ template <>
 struct AssignOp_Impl<AssignOp::Set>
 {
    template <typename lvalue_t, typename rvalue_t>
-   MFEM_HOST_DEVICE
    static inline lvalue_t &Assign(lvalue_t &a, const rvalue_t &b)
    {
       return (a = b);
    }
+
+   template <typename lvalue_t, typename rvalue_t>
+   MFEM_HOST_DEVICE
+   static inline lvalue_t &AssignHD(lvalue_t &a, const rvalue_t &b)
```

Comment:
Why did you needed to duplicate these `Assign` functions with the `HD` postfix?
I'm not a big fan of `HD` as a postfix too.
`hd_`, `h_d_` prefix?

Reply:
This is because of how `nvcc` instantiates `__host__ __device__` methods in templates: if one version (host or device) is needed then the other one is also instantiated. This was producing to a ton of warnings when used with the SIMD types -- about calling host functions from device+host functions. I tried using pragma to suppress the warnings but that lead to compiler internal error in `nvcc`. This solution is not ideal (we need to copy-paste the source twice) but I couldn't think of a more elegant solution. We maybe able to duplicate the code with a macro but I didn't look into that.

Reply:
`discussed`